pylutron_caseta/pairing.py

```diff
@@ -204,16 +205,22 @@ def _generate_csr_with_ssl_context():
             ssl_context.load_verify_locations(cadata=LAP_CA_PEM)
             ssl_context.load_cert_chain(lap_cert_temp_file.name, lap_key_temp_file.name)
             ssl_context.verify_mode = ssl.CERT_REQUIRED
+            
+            if (IS_WINDOWS):
+                lap_key_temp_file.close()
```

Comment:
I don't think this will solve your problem. The file needs to be closed before the call to `ssl_context.load_cert_chain`. That's really the problem with NamedTemporaryFile. Closing the file so it can be opened for reading again causes it to be deleted.

Reply:
@mdonoughe  it seems to be working fine being open until that point, though I don't have any other ways of testing it at the moment so I don't know if it's just me

Reply:
I'd expect this code to work the same as the original code unless setting delete has some interaction with file share modes.

Reply:
You are right. I guess there is an interaction with file sharing.

Reply:
@mdonoughe just did a little more testing, and there doesn't appear to be a difference calling it before or after ` ssl_context.load_cert_chain` I'll leave it before just incase it's just me.

Reply:
Actually you probably want to leave it after if that works. Otherwise the error handling becomes more complicated because the file always needs to be closed before deletion. In any case, the handling for the two files should be consistent.